docs/concepts/services-networking/dns-pod-service.md

```diff
@@ -262,157 +199,97 @@ spec:
   dnsPolicy: ClusterFirstWithHostNet
 ```
 
-#### Quick diagnosis
+### Pod's DNS Config
 
-Errors such as the following indicate a problem with the kube-dns add-on or associated Services:
+Kubernetes v1.9 introduces an Alpha feature that allows users more control on
+the DNS settings for a Pod. To enable this feature, the cluster aministrator
+needs to enable the `CustomPodDNS` feature gate on the apiserver and the kubelet,
+for example, "`--feature-gates=CustomPodDNS=true,...`".
+When the feature gate is enabled, users can set the `dnsPolicy` field of a Pod
+to "`None`" and they can add a new field `dnsConfig` to a Pod Spec.
 
-```
-$ kubectl exec -ti busybox -- nslookup kubernetes.default
-Server:    10.0.0.10
-Address 1: 10.0.0.10
+The `dnsConfig` field is optional and it can work with any `dnsPolicy` settings.
+However, when a Pod's `dnsPolicy` is set to "`None`", the `dnsConfig` field has
+to be specified.
 
-nslookup: can't resolve 'kubernetes.default'
-```
+Below are the properties a user can specify in the `dnsConfig` field:
 
-or
+- `nameservers`: a list of IP addresses that will be used as DNS servers for the
+  Pod. There can be at most 3 IP addresses specified. When the Pod's `dnsPolicy`
+  is set to "`None`", the list must contain at least one IP address, otherwise
+  this property is optional.
+  The servers listed will be combined to the base nameservers generated from the
+  specified DNS policy with duplicate addresses removed.
+- `searches`: a list of DNS search domains for hostname lookup in the Pod.
+  This property is optional. When specified, the provided list will be merged
+  into the base search domain names generated from the chosen DNS policy.
+  Duplicate domain names are removed.
+  Kubernetes allows for at most 6 search domains.
+- `options`: an optional list of objects where each object may have a `name`
+  property (required) and a `value` property (optional). The contents in this
+  property will be merged to the options generated from the specified DNS policy.
+  Duplicate entries are removed.
 
-```
-$ kubectl exec -ti busybox -- nslookup kubernetes.default
-Server:    10.0.0.10
-Address 1: 10.0.0.10 kube-dns.kube-system.svc.cluster.local
+The following is an example Pod with custom DNS settings:
 
-nslookup: can't resolve 'kubernetes.default'
-```
+{% include code.html language="yaml" file="custon-dns.yaml" ghlink="/docs/concepts/services-networking/custom-dns.yaml" %}
 
-#### Check if the DNS pod is running
-
-Use the kubectl get pods command to verify that the DNS pod is running.
+When the Pod above is created, the container `test` gets the following contents
+in its `/etc/resolv.conf` file:
 
 ```
-kubectl get pods --namespace=kube-system -l k8s-app=kube-dns
+nameserver 1.2.3.4
+search ns1.svc.cluster.local my.dns.search.suffix
+options ndots:2 edns0
 ```
 
-You should see something like:
+## Compatibility
```

Comment:
SIG-docs supports documentation for the current release and the four previous releases. With the release of 1.9, our oldest release would be 1.5.

Is there anything in this section that applies to version compatibility for >1.5? If not, you can safely remove this entire section.